app/js/filterFunctions.js

```diff
@@ -98,3 +102,27 @@ function percentageFilter (resultArray){
     }
     return newResultArray
 }
+
+/**
+ * This function filters the list of users to be displayed in the table based on the test
+ * allocated to that user
+ * 
+ * @param {resultArray} Array array of users to be filtered
+ * 
+ * @return Array the filtered array of users
+ */
+function testAllocatedFilter(resultArray) {
+    let filterTestAllocated = document.getElementById('testAllocated')
+    let newResultArray = []
+    let chosenTestAllocated = filterTestAllocated.value
+    if (chosenTestAllocated != "") {
+        resultArray.forEach((data)=>{
+            if(data.testAllocated == chosenTestAllocated) {
+                newResultArray.push(data)
+            }
+        })
+    } else {
+        return resultArray
+    }
+    return newResultArray
```

Comment:
The logic structure here can be simplified by using what's called an early return. I don't want to directly give the code away but you can eliminate the else by moving your return of newResultArray to be elsewhere. Speak to me if you have problems!